src/util/cursor.js

```diff
@@ -0,0 +1,93 @@
+const _ = require('lodash')
+const { encodeBase64, decodeBase64 } = require('./encoding')
+
+const allowedTypes = [
+  'number',
+  'boolean',
+  'string',
+
+  // Date instances are returned from database for timestamptz type
+  // which is different from date stored in string type
+  'date',
+]
+
+// must be uncommon chain of characters to distinguish
+// from real string values
+const separator = '~|~'
```

Comment:
Yes indeed. Your suggestion is a good one.
This separator might break if this exact value is used by users (even if it's unlikely).
The only drawback will be the cursor length but it may be worth.

Reply:
It’s an implementation detail hidden from interface, right (even when not using our JS SDK)? Which would mean no breaking change in the future (provided we detect and handle both formats).

In this case maybe we can keep this as it is for now and refactor if needed in the future.

Reply:
Yes indeed. We can change the cursor format in the future with backward compatibility.